app/src/test/java/org/fossasia/openevent/app/presenter/LoginPresenterTest.java

```diff
@@ -136,4 +143,39 @@ public void shouldNotAccessView() {
         Mockito.verifyNoMoreInteractions(loginView);
     }
 
+    @Test
+    public void shouldAttachEmailOnLoginSuccessfully() {
+        String authToken = "testToken";
+
+        Mockito.when(loginModel.login(email, password))
+            .thenReturn(Observable.just(new LoginResponse(authToken)));
+
+        Mockito.when(utilModel.getStringSet(Constants.SHARED_PREFS_SAVED_EMAIL, null))
+            .thenReturn(savedEmails);
+
+        loginPresenter.attach(loginView);
+        loginPresenter.login(email, password);
+
+        Mockito.verify(loginView).attachEmails(savedEmails);
+    }
+
+    @Test
+    public void shouldAttachEmailAutomatically() {
+        Mockito.when(utilModel.getStringSet(Constants.SHARED_PREFS_SAVED_EMAIL, null)).thenReturn(savedEmails);
+        Mockito.when(loginModel.isLoggedIn()).thenReturn(false);
+
+        loginPresenter.attach(loginView);
+
+        Mockito.verify(loginView).attachEmails(savedEmails);
+    }
+
+    @Test
+    public void shouldNotAttachEmailAutomatically() {
+        Mockito.when(utilModel.getStringSet(Constants.SHARED_PREFS_SAVED_EMAIL, null)).thenReturn(null);
+        Mockito.when(loginModel.isLoggedIn()).thenReturn(false);
+
+        loginPresenter.attach(loginView);
+
+        Mockito.verifyZeroInteractions(loginView);
```

Comment:
Make this Mockito.verify(loginView, Mockito.never()).attachEmails();

Because we may want to add other interactions in future which will fail these tests.

Other than that, eveything's fine